### ISSUE TITLE
Fallback dropdown menus can be displayed if Dashboard fails to fetch version

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#16610]
**Setup**
- Rancher version: 2.14-head
- Rancher UI Extensions: NA
- Browser type & version: Firefox/Chrome

**Describe the bug**

The dropdown menu fallback versions can be rendered if Rancher Dashboard fails to fetch the version setting.

**To Reproduce**

1. Delete the `server-version` setting
   2. Navigate to Local Cluster Explorer and open the Kubectl Shell
   3. Run the following command to delete the setting
      ```
      kubectl delete setting server-version
      ```
   4. Confirm that the setting is deleted
      ```
      > kubectl get setting server-version -o json
      Error from server (NotFound): settings.longhorn.io "server-version" not found
      ```
2. Navigate to Local Cluster Explorer => Workloads => Deployments
3. In the Namespace Filter, select "All Namespaces"

**Result**

The overflow buttons for table actions are the fallback versions, lacking accessibility enhancements.

**Expected Result**

The overflow buttons for table actions are the modern version with accessibility enhancements.

**Screenshots**

NA

**Additional context**

> 🗒️ NOTE: This has only been observed in some settings. I haven't been able to reproduce this issue reliably without manipulating settings myself

**Comment by rak-phillip** [comment]
@gaktive @nwmac this could be a good backport candidate for v2.13.3

**Comment by a-arias** [comment]
**To Test**
Fix validated ✅
Rancher :v2.14-62061d2656fb02d5acba3cad5d7b61a61002342c-head
UI:master f78e32e

https://github.com/user-attachments/assets/0f4b59de-0cc1-42f4-9bf5-cd130bf0a5d9
